src/client/Client.cc

```diff
@@ -11584,8 +11584,12 @@ int Client::_do_setxattr(Inode *in, const char *name, const void *value,
 {
 
   int xattr_flags = 0;
-  if (!value)
-    xattr_flags |= CEPH_XATTR_REMOVE;
+  // If the size is set to 0, set the value to an empty string
+  if(size == 0) {
+    value = "";
+  } else if(size != 0 && value == NULL) {
```

Comment:
Space between the if and '(' here too. Also, you don't need to check size here since you know that it's non-zero.

Reply:
Yea sorry about that. Fixed it.